code/game/objects/items/robot/robot_upgrades.dm

```diff
@@ -545,29 +545,30 @@
 
 /obj/item/borg/upgrade/expand/action(mob/living/silicon/robot/robot, user = usr)
 	. = ..()
-	if(.)
+	if(!.)
+		return FALSE
 
-		if(robot.hasExpanded)
-			to_chat(usr, span_warning("This unit already has an expand module installed!"))
-			return FALSE
+	if(robot.hasExpanded)
+		to_chat(usr, span_warning("This unit already has an expand module installed!"))
+		return FALSE
 
-		robot.notransform = TRUE
-		var/prev_lockcharge = robot.lockcharge
-		robot.SetLockdown(TRUE)
-		robot.set_anchored(TRUE)
-		var/datum/effect_system/fluid_spread/smoke/smoke = new
-		smoke.set_up(1, holder = robot, location = robot.loc)
-		smoke.start()
-		sleep(0.2 SECONDS)
-		for(var/i in 1 to 4)
-			playsound(robot, pick('sound/items/drill_use.ogg', 'sound/items/jaws_cut.ogg', 'sound/items/jaws_pry.ogg', 'sound/items/welder.ogg', 'sound/items/ratchet.ogg'), 80, TRUE, -1)
-			sleep(1.2 SECONDS)
-		if(!prev_lockcharge)
-			robot.SetLockdown(FALSE)
-		robot.set_anchored(FALSE)
-		robot.notransform = FALSE
-		robot.hasExpanded = TRUE
-		robot.update_transform(2)
+	ADD_TRAIT(robot, TRAIT_NO_TRANSFORM, REF(src))
```

Comment:
Unrelated to your code but does this upgrade really add notransform but not sanity check for it? Lol, you could probably exploit this somehow

Reply:
yeah i don't think it blocks anywhere else, maybe it was just meant to give the cosmetic "can't do shit" thing?